network.go

```diff
@@ -74,10 +73,10 @@ const (
 	defaultQueues     = 8
 )
 
-// NetIfaceAddrs describes a network interface an its associated IP addresses.
+// NetIfaceAddrs describes a network interface and its associated IP addresses.
 type NetIfaceAddrs struct {
-	Iface net.Interface
-	Addrs []net.IPNet
+	Iface netlink.LinkAttrs
```

Comment:
@sboeuf I think it is better to actually have the actual link vs the LinkAttrs. The reason is that the link will help us get the interface type. If you only cache the attrs we lose information that we need.

Reply:
Done